src/modules/common/popup/popup.component.js

```diff
@@ -1,38 +1,199 @@
-import React from 'react';
+import React, { Component } from 'react';
+import $ from 'jquery';
 import PropTypes from 'prop-types';
+import Icon from '../icon/icon';
 
 import './css/popup.component.css';
 
-const PopupComponent = (props) => {
-    const attrs = {
-        className: 'c-popup',
-        hidden: !props.visible,
-    };
-
-    return (
-        <div {...attrs}>
-            <div className="c-popup__header">
-                <div className="c-popup__title">{props.title}</div>
-                <div className="c-popup__close" onClick={props.onClose}>
-                    <svg className="ez-icon">
-                        <use xlinkHref="/bundles/ezplatformadminui/img/ez-icons.svg#discard" />
-                    </svg>
+const CLASS_NON_SCROLLABLE = 'ezs-non-scrollable';
+const CLASS_MODAL_OPEN = 'modal-open';
+const MODAL_CONFIG = {
+    backdrop: 'static',
+    keyboard: true,
+};
+const MODAL_SIZE_CLASS = {
+    small: 'modal-sm',
+    medium: '',
+    large: 'modal-lg',
+};
+
+class Popup extends Component {
+    constructor(props) {
+        super(props);
+
+        this._refModal = null;
+
+        this.setModalRef = this.setModalRef.bind(this);
+        this.onKeyUp = this.onKeyUp.bind(this);
+
+        this.state = { isVisible: props.isVisible, isLoading: props.isLoading };
+    }
+
+    componentDidMount() {
+        const { isVisible: show } = this.state;
+
+        if (show) {
+            $(this._refModal).modal({ ...MODAL_CONFIG, show, focus: this.props.hasFocus });
+
+            this.attachModalEventHandlers();
+        }
+    }
+
+    componentDidUpdate() {
+        const { isVisible: show } = this.state;
+
+        $(this._refModal).modal({ ...MODAL_CONFIG, show, focus: this.props.hasFocus });
+
+        if (show) {
+            this.attachModalEventHandlers();
+        }
+    }
+
+    componentWillUnmount() {
+        $(this._refModal).modal('hide');
+        document.body.classList.remove(CLASS_MODAL_OPEN, CLASS_NON_SCROLLABLE);
+    }
+
+    UNSAFE_componentWillReceiveProps({ isVisible, onConfigIframeLoad, isLoading }) {
+        this.setState((state) => ({ ...state, isVisible, onConfigIframeLoad, isLoading }));
+    }
+
+    attachModalEventHandlers() {
+        const modal = $(this._refModal);
+
+        modal.on('keyup', this.onKeyUp);
+        modal.one('hidden.bs.modal', this.props.onClose);
+    }
+
+    onKeyUp(event) {
+        const { originalEvent } = event;
+        const escKeyCode = 27;
+        const escKeyPressed = originalEvent && (originalEvent.which === escKeyCode || originalEvent.keyCode === escKeyCode);
+
+        if (escKeyPressed) {
+            this.props.onClose();
+        }
+    }
+
+    setModalRef(component) {
+        this._refModal = component;
+    }
+
+    renderHeader() {
+        const closeBtnLabel = Translator.trans(/*@Desc("Close")*/ 'popup.close.label', {}, 'universal_discovery_widget');
```

Comment:
ping @sunpietro @dew326 Do we want to have separate translations domain for `common` components?

Reply:
We have this one! 😄

Reply:
It's probably the first and the only one, for now. :-)